test/cases/ostree-vsphere.sh

```diff
@@ -0,0 +1,884 @@
+#!/bin/bash
+
+# This test scrip covers:
+# 1. build edge-container image, build edge-vsphere image
+# 2. upload image to vsphere datacenter7.0-intel, create edge vm, run test cases against vm.
+# 3. upload image to vsphere datacenter7.0-amd, create edge vm, run test cases against vm.
+# 4. build upgrade edge-container image
+# 5. run rpm-ostree upgrade in all edge vms, then run test cases against upgraded vms.
+# 6. delete all vsphere vms, images
+
+set -euo pipefail
+
+# Provision the software under test.
+/usr/libexec/osbuild-composer-test/provision.sh none
+
+# Get OS data.
+source /etc/os-release
+ARCH=$(uname -m)
+
+source /usr/libexec/tests/osbuild-composer/shared_lib.sh
+
+# Install sshpass
+sudo dnf install -y sshpass
+
+# Install ansible module
+sudo ansible-galaxy collection install community.vmware
+sudo pip3 install PyVmomi
+
+# Install govc
+sudo curl -L -o - "https://github.com/vmware/govmomi/releases/latest/download/govc_$(uname -s)_$(uname -m).tar.gz" | tar -C /usr/local/bin -xvzf - govc
+
+# Start firewall
+sudo systemctl enable --now firewalld
+
+# Start libvirtd and test it.
+greenprint "🚀 Starting libvirt daemon"
+sudo systemctl start libvirtd
+sudo virsh list --all > /dev/null
+
+# Set a customized dnsmasq configuration for libvirt so we always get the
+# same address on bootup.
+sudo tee /tmp/integration.xml > /dev/null << EOF
+<network xmlns:dnsmasq='http://libvirt.org/schemas/network/dnsmasq/1.0'>
+  <name>integration</name>
+  <uuid>1c8fe98c-b53a-4ca4-bbdb-deb0f26b3579</uuid>
+  <forward mode='nat'>
+    <nat>
+      <port start='1024' end='65535'/>
+    </nat>
+  </forward>
+  <bridge name='integration' zone='trusted' stp='on' delay='0'/>
+  <mac address='52:54:00:36:46:ef'/>
+  <ip address='192.168.100.1' netmask='255.255.255.0'>
+    <dhcp>
+      <range start='192.168.100.2' end='192.168.100.254'/>
+      <host mac='34:49:22:B0:83:30' name='vm' ip='192.168.100.50'/>
+    </dhcp>
+  </ip>
+  <dnsmasq:options>
+    <dnsmasq:option value='dhcp-vendorclass=set:efi-http,HTTPClient:Arch:00016'/>
+    <dnsmasq:option value='dhcp-option-force=tag:efi-http,60,HTTPClient'/>
+    <dnsmasq:option value='dhcp-boot=tag:efi-http,&quot;http://192.168.100.1/httpboot/EFI/BOOT/BOOTX64.EFI&quot;'/>
+  </dnsmasq:options>
+</network>
+EOF
+
+if ! sudo virsh net-info integration > /dev/null 2>&1; then
+    sudo virsh net-define /tmp/integration.xml
+fi
+if [[ $(sudo virsh net-info integration | grep 'Active' | awk '{print $2}') == 'no' ]]; then
+    sudo virsh net-start integration
+fi
+
+# Allow anyone in the wheel group to talk to libvirt.
```

Comment:
There's no VM created in this case. This part of code should be removed.

Reply:
Removed